core/src/main/java/cucumber/runtime/Runtime.java

```diff
@@ -210,6 +212,31 @@ public Runtime build() {
         }
     }
 
+    private static final class CucumberThreadFactory implements ThreadFactory {
+
+        private final AtomicInteger threadNumber = new AtomicInteger(1);
+        private final ThreadGroup group;
+
+        CucumberThreadFactory() {
+            SecurityManager s = System.getSecurityManager();
+            this.group = s != null ? s.getThreadGroup() : Thread.currentThread().getThreadGroup();
```

Comment:
I don't think it's required nor desirable to manually set the thread group. The `ThreadGroup` for the workers should be the same as the `ThreadGroup` of `Runtime.run`. Currently it is the `ThreadGroup` of `Runtime.Builder.build` which need not be the same. Using a different constructor will automatically use the correct `ThreadGroup`.